src/Microsoft.VisualStudio.Threading/ReentrantSemaphore.cs

```diff
@@ -143,7 +142,12 @@ public static ReentrantSemaphore Create(int initialCount = 1, JoinableTaskContex
         /// <summary>
         /// Executes a given operation within the semaphore.
         /// </summary>
-        /// <param name="operation">The delegate to invoke once the semaphore is entered.</param>
+        /// <param name="operation">
+        /// The delegate to invoke once the semaphore is entered. Not guaranteed to be executed on the context this was originally called on.
```

Comment:
> Not guaranteed to be executed on the context this was originally called on. [](start = 65, length = 76)

I would drop this sentence, since it *is* guaranteed in some conditions, which you fully describe after this sentence.